cmd/podman/volumes/export.go

```diff
@@ -0,0 +1,78 @@
+package volumes
+
+import (
+	"context"
+
+	"github.com/containers/common/pkg/completion"
+	"github.com/containers/podman/v3/cmd/podman/common"
+	"github.com/containers/podman/v3/cmd/podman/inspect"
+	"github.com/containers/podman/v3/cmd/podman/registry"
+	"github.com/containers/podman/v3/pkg/domain/entities"
+	"github.com/containers/podman/v3/utils"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	volumeExportDescription = `
+podman volume export
+
+Allow content of volume to be exported into external tar.`
+	exportCommand = &cobra.Command{
+		Annotations:       map[string]string{registry.EngineMode: registry.ABIMode},
+		Use:               "export [options]",
+		Short:             "Export volumes",
+		Args:              cobra.ExactArgs(1),
+		Long:              volumeExportDescription,
+		RunE:              export,
+		ValidArgsFunction: common.AutocompleteVolumes,
+	}
+)
+
+var (
+	// Temporary struct to hold cli values.
+	cliExportOpts = struct {
+		Output string
+	}{}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: exportCommand,
+		Parent:  volumeCmd,
+	})
+	flags := exportCommand.Flags()
+
+	outputFlagName := "output"
+	flags.StringVarP(&cliExportOpts.Output, outputFlagName, "o", "/dev/stdout", "Output=[path]")
+	_ = exportCommand.RegisterFlagCompletionFunc(outputFlagName, completion.AutocompleteDefault)
+}
+
+func export(cmd *cobra.Command, args []string) error {
+	var inspectOpts entities.InspectOptions
+	containerEngine := registry.ContainerEngine()
+	ctx := context.Background()
+
+	if cliExportOpts.Output == "" {
+		return errors.New("expects output path, use --output=[path]")
+	}
+	inspectOpts.Type = inspect.VolumeType
+	volumeData, _, err := containerEngine.VolumeInspect(ctx, args, inspectOpts)
+	if err != nil {
+		return err
+	}
+	if len(volumeData) < 1 {
+		return errors.New("no volume data found")
+	}
+
+	mountPoint := volumeData[0].VolumeConfigResponse.Mountpoint
```

Comment:
This will only work on volumes that do not require a mount. On volumes using a volume plugin, or volumes using the local plugin with options that require a filesystem mount, we have no guarantee that the volume has been mounted, so this directory could easily be empty.

Reply:
@mheon oh but if its filesystem mount or a plugin and not mounted on the mountpoint container wouldnt see the content either right ? correct me if i am wrong.  So we only extract the content which is visible to container via this volume. But let me verify it.

Reply:
We should be able to view the content on the host, that's where the mounts happen.

Reply:
Maybe it would be better to temporarily block export on volumes that require mount? There's a helper function to determine that.

Reply:
ah let me check it.

Reply:
@mheon So i checked eventually the mountPoint is coming from state ( https://github.com/containers/podman/blob/main/libpod/volume.go#L143 )  there might be chances that this just might work so instead of skipping I was thinking shouldnt we just try exporting and add a note in manpage that `export does not guarantees when the driver type is external plugin and its depends upon driver configurations`.

Reply:
We should at least check if the volume is mounted - succeeding when we copied nothing since the volume was unmounted would be bad.

Reply:
Yes report an erro rif the volume is not mounted and a mount is necessary.